sirepo/package_data/static/js/sirepo-components.js

```diff
@@ -3253,60 +3251,62 @@ SIREPO.app.directive('jobsList', function(requestSender, appState, $location, $s
                 );
             }
 
-            $scope.endSimulation = function(rowIndex) {
-                let s = $scope.data.rows[rowIndex][1];
-                let r = $scope.data.rows[rowIndex][7];
-                let t = $scope.data.rows[rowIndex][0];
-                let j = {
-                    request: {simulationId: s, report: r, simulationType: t}
+            $scope.displayedCols = $scope.wantAdm ? [
+                'simulationType',
+                'simulationId',
+                'uid',
+                'startTime',
+                'lastUpdateTime',
+                'elapsedTime',
+                'statusMessage',
+                'queuedTime',
+                'driverDetails',
+                'isPremiumUser'
+            ] : [
+                'startTime',
+                'lastUpdateTime',
+                'elapsedTime',
+                'statusMessage'
+            ];
+
+            $scope.endSimulation = function(job) {
+                const r = {
+                    simulationId: job.simulationId,
+                    report: job.computeModel,
+                    simulationType: job.simulationType,
                 };
-                let successCallback = () => {
+                const successCallback = () => {
                     $scope.getJobs();
                 };
-                let errorCallback = () => {
-                    srlog('Error with runCancel request');
+                const errorCallback = (request) => {
+                    return (data) => {
+                        srlog('runCancel error=${data.error} from request=${request}');
+                    };
                 };
-                requestSender.sendRequest('runCancel', successCallback, j.request, errorCallback);
+                requestSender.sendRequest('runCancel', successCallback, r, errorCallback(r));
             };
 
-            $scope.getCellContent = function(rowIndex, colIndex) {
-                let row = $scope.data.rows[rowIndex];
-                if (! row[colIndex] || colIndex === row.length - 1) {
-                    return;
-                }
+            $scope.getCellContent = function(job, key, colIndex) {
                 const typeDispatch = {
                     DateTime: appState.formatDate,
                     Time: appState.formatTime,
                     String: function(s){return s;},
                 };
-
-                let cellData = $scope.data.rows[rowIndex][colIndex];
-                let typeDispatchFunction = typeDispatch[$scope.data.header[colIndex][1]];
-                return typeDispatchFunction(cellData);
+                return typeDispatch[$scope.data.header[colIndex + getStartIndex() + ($scope.wantAdm ? 0 : 1)][1]](job[key]);
```

Comment:
Since job is no longer an array I think header doesn't have to be one either. Then we won't have to do this funky indexing stuff. I think header can be something like
```python
header = PKDict(
  simulationId=PKDict(
    title='Simulation Id'
    type='String'
  )
# all the other fields...
)
```

Then you can access the desired header field just based on the name of the field (ex simulationId)

Reply:
a18414df57002245600d5d988578afea651d41d1